doc/source/developing/releasing.rst

```diff
@@ -1,37 +1,41 @@
 How to Do a Release
 -------------------
 
-Periodically, the yt development community issues new releases. Since yt follows
-`semantic versioning <https://semver.org/>`_, the type of release can be read off
+Periodically, the yt development community issues new releases. yt loosely follows
+`semantic versioning <https://semver.org/>`_. The type of release can be read off
 from the version number used. Version numbers should follow the scheme
 ``MAJOR.MINOR.PATCH``. There are three kinds of possible releases:
 
 * Bugfix releases
 
-  These releases are regularly scheduled and will optimally happen approximately
-  once a month. These releases should contain only fixes for bugs discovered in
+  These releases should contain only fixes for bugs discovered in
   earlier releases and should not contain new features or API changes. Bugfix
-  releases should increment the ``PATCH`` version number. Bugfix releases should
+  releases only increment the ``PATCH`` version number. Bugfix releases should
   *not* be generated by merging from the ``main`` branch, instead bugfix pull
-  requests should be manually backported. Version ``3.2.2`` is a bugfix release.
+  requests should be backported to a dedicated branch.
+  See :ref:`doing-a-bugfix-release`. Version ``3.2.2`` is a bugfix release.
 
 * Minor releases
 
   These releases happen when new features are deemed ready to be merged into the
   ``stable`` branch and should not happen on a regular schedule. Minor releases
   can also include fixes for bugs if the fix is determined to be too invasive
-  for a bugfix release. Minor releases should *not* include
-  backwards-incompatible changes and should not change APIs.  If an API change
```

Comment:
I think this sentence is important to keep.

Reply:
It is not consistent with the undergoing evolutions. We scheduled backwards incompatible changes in 4.1 and 4.2, but all of them are preceded by clear deprecation warnings.

Reply:
We should not encourage major API changes in minor versions, but there are exceptions when API changes are necessary. Communicating that clearly in the release docs is important because not everybody looks at our governance. 

Are you proposing a change in our release structure?

Reply:
I'm not. But "major" is a relative term, while "backward incompatible" isn't. 

IMHO, expired deprecations are not necessarily "major" changes, but they are backward incompatible.

If we were strictly following semantic versioning, as currently suggested by the docs, yt 4.1 should be numbered 5.0

Reply:
Right. Major changes are not expired deprecations. Expired deprecations are the "deemed necessary" API change that the next sentence addresses. By removing this sentence completely without revising it we are not discouraging API changes from minor version to minor version, which is not a good practice.

We should also not be deprecating functions unless really necessary from minor version to minor version. Our users are the reason the project is successful, and we need to weigh whether breaking their scripts in a minor version is urgently necessary or if it can wait until a major version.

Reply:
> We should also not be deprecating functions unless really necessary from minor version to minor version.

kinda +1/2 here. IMO deprecations should appear as soon as there's a better :tm: alternative available. But I agree that _expiring_ deprecations between minor versions is not to be encouraged (this was my motivation for #3603).

> Major changes are not expired deprecations.

Thank you for clarifying this, then we're actually on the same page I think ?
How can I rephrase this to be more discouraging ? Keeping the existing phrasing still seems inadequate because it's de facto not what we're doing.

Reply:
@munkm how about this
"Other than expired old deprecations, minor releases should *not* include backward incompatible changes."